src/phases/select-modifier-phase.ts

```diff
@@ -98,259 +80,300 @@ export class SelectModifierPhase extends BattlePhase {
+          });
         } else {
-          const pokemonModifierType = modifierType as PokemonModifierType;
-          const isMoveModifier = modifierType instanceof PokemonMoveModifierType;
-          const isTmModifier = modifierType instanceof TmModifierType;
-          const isRememberMoveModifier = modifierType instanceof RememberMoveModifierType;
-          const isPpRestoreModifier =
-            modifierType instanceof PokemonPpRestoreModifierType || modifierType instanceof PokemonPpUpModifierType;
-          const partyUiMode = isMoveModifier
-            ? PartyUiMode.MOVE_MODIFIER
-            : isTmModifier
-              ? PartyUiMode.TM_MODIFIER
-              : isRememberMoveModifier
-                ? PartyUiMode.REMEMBER_MOVE_MODIFIER
-                : PartyUiMode.MODIFIER;
-          const tmMoveId = isTmModifier ? (modifierType as TmModifierType).moveId : undefined;
-          globalScene.ui.setModeWithoutClear(
-            UiMode.PARTY,
-            partyUiMode,
-            -1,
-            (slotIndex: number, option: PartyOption) => {
-              if (slotIndex < 6) {
-                globalScene.ui.setMode(UiMode.MODIFIER_SELECT, this.isPlayer()).then(() => {
-                  const modifier = !isMoveModifier
-                    ? !isRememberMoveModifier
-                      ? modifierType.newModifier(party[slotIndex])
-                      : modifierType.newModifier(party[slotIndex], option as number)
-                    : modifierType.newModifier(party[slotIndex], option - PartyOption.MOVE_1);
-                  applyModifier(modifier!, true); // TODO: is the bang correct?
-                });
-              } else {
-                globalScene.ui.setMode(
-                  UiMode.MODIFIER_SELECT,
-                  this.isPlayer(),
-                  this.typeOptions,
-                  modifierSelectCallback,
-                  this.getRerollCost(globalScene.lockModifierTiers),
-                );
-              }
-            },
-            pokemonModifierType.selectFilter,
-            modifierType instanceof PokemonMoveModifierType
-              ? (modifierType as PokemonMoveModifierType).moveSelectFilter
-              : undefined,
-            tmMoveId,
-            isPpRestoreModifier,
-          );
+          this.resetModifierSelect(modifierSelectCallback);
         }
+      },
+      pokemonModifierType.selectFilter,
+      modifierType instanceof PokemonMoveModifierType
+        ? (modifierType as PokemonMoveModifierType).moveSelectFilter
+        : undefined,
+      tmMoveId,
+      isPpRestoreModifier,
+    );
+  }
+
+  // Function that determines how many reward slots are available
+  private getModifierCount(): number {
+    const modifierCountHolder = new NumberHolder(3);
+    if (this.isPlayer()) {
+      globalScene.applyModifiers(ExtraModifierModifier, true, modifierCountHolder);
+      globalScene.applyModifiers(TempExtraModifierModifier, true, modifierCountHolder);
+    }
```

Comment:
If there's a `this.isPlayer()` check it should be done at the top of `start()` and end the phase early if somehow things are *that* screwed up that a `SelectModifierPhase` was pushed for an enemy.
```suggestion
    globalScene.applyModifiers(ExtraModifierModifier, true, modifierCountHolder);
    globalScene.applyModifiers(TempExtraModifierModifier, true, modifierCountHolder);
```